azure-client-runtime/src/main/java/com/microsoft/azure/v2/AzureProxy.java

```diff
@@ -182,39 +182,51 @@ public static HttpPipeline defaultPipeline(Class<?> swaggerInterface, RequestPol
+     * @param <A> The type of the Swagger interface.
+     * @return A proxy implementation of the provided Swagger interface.
+     */
+    @SuppressWarnings("unchecked")
+    public static <A> A create(Class<A> swaggerInterface, AzureServiceClient azureServiceClient) {
+        return AzureProxy.create(swaggerInterface, azureServiceClient.azureEnvironment(), azureServiceClient.httpPipeline(), azureServiceClient.serializerAdapter());
+    }
+
+    /**
+     * Create a proxy implementation of the provided Swagger interface.
+     * @param swaggerInterface The Swagger interface to provide a proxy implementation for.
      * @param httpPipeline The HTTP httpPipeline will be used to make REST calls.
      * @param serializer The serializer that will be used to convert POJOs to and from request and
      *                   response bodies.
      * @param <A> The type of the Swagger interface.
      * @return A proxy implementation of the provided Swagger interface.
      */
     @SuppressWarnings("unchecked")
-    public static <A> A create(Class<A> swaggerInterface, AzureEnvironment azureEnvironment, HttpPipeline httpPipeline, SerializerAdapter<?> serializer) {
-        String baseUrl = null;
-
-        if (azureEnvironment != null) {
-            final AzureHost azureHost = swaggerInterface.getAnnotation(AzureHost.class);
-            if (azureHost != null) {
-                baseUrl = azureEnvironment.url(azureHost.endpoint());
-            }
-        }
-
-        return AzureProxy.create(swaggerInterface, baseUrl, httpPipeline, serializer);
+    public static <A> A create(Class<A> swaggerInterface, HttpPipeline httpPipeline, SerializerAdapter<?> serializer) {
+        return AzureProxy.create(swaggerInterface, null, httpPipeline, serializer);
     }
 
     /**
      * Create a proxy implementation of the provided Swagger interface.
      * @param swaggerInterface The Swagger interface to provide a proxy implementation for.
-     * @param baseUrl The base URL (protocol and host) that the proxy implementation will target.
-     * @param httpPipeline The internal HTTP httpPipeline that will be used to make REST calls.
+     * @param azureEnvironment The azure environment that the proxy implementation will target.
+     * @param httpPipeline The HTTP httpPipeline will be used to make REST calls.
      * @param serializer The serializer that will be used to convert POJOs to and from request and
      *                   response bodies.
      * @param <A> The type of the Swagger interface.
      * @return A proxy implementation of the provided Swagger interface.
      */
     @SuppressWarnings("unchecked")
-    public static <A> A create(Class<A> swaggerInterface, String baseUrl, HttpPipeline httpPipeline, SerializerAdapter<?> serializer) {
+    public static <A> A create(Class<A> swaggerInterface, AzureEnvironment azureEnvironment, HttpPipeline httpPipeline, SerializerAdapter<?> serializer) {
```

Comment:
Are you changing the generator to pass AzureEnvironment down so that RestProxy can do the lookup here? Also, was it basically decided that providing the base URL directly is unnecessary for any Azure use case?

Reply:
Providing a baseUrl is unnecessary now. We have both the Host and AzureHost annotations which provide all of the functionality that baseUrl used to provide.
RestProxy never needs to know about baseUrl. Only SwaggerInterfaceParser needs the baseUrl, and it gets that from either AzureProxy (parsing the AzureHost annotation) or from its own parsing of the Host annotation.